### PR TITLE
add softmax to jax

### PR DESCRIPTION
**Pull request by PatricYan**
- [ ] #3015  
1. add softmax to jax
2. test case

### CODE REVIEW

**Comment by PatricYan** [comment]
as newly committed code merged, I resolved the conflict, then push it

**Comment by jkeane508** [comment]
Once checks have completed, I will review and request changes if need be :)

**Comment by PatricYan** [comment]
> Once checks have completed, I will review and request changes if need be :)

ok, checks have been completed. would you like to make a review for it

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="jax",
+        fn_tree="nn.softmax",
+        x=data,
+        axis=data.ndim - 1,
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:40]: This is not the best way to test this function, we should test all possible axis available.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+import ivy.functional.backends.jax as ivy_jax
+
+
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=ivy_jax.valid_numeric_dtypes, min_num_dims=2, min_dim_size=3
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:12]: We should be able to test all possible array dimensions

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hi, just some changes in regards to the testing data. We should be able to test arrays of all dimensions for this function.

**Comment by PatricYan** [comment]
Hi, this is not **softmax** test case error, it is **gelu** test case error as the backend code has been updated.

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="jax",
+        fn_tree="nn.softmax",
+        x=data,
+    )
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:42]: We should be testing the axis parameter. We can generate this with `axis=helpers.ints(min_value=-1, max_value=0)` as done with the relevant backend test [here](https://github.com/unifyai/ivy/blob/971fba4acb10703b36b4d21aa5adeca3d09cc75d/ivy_tests/test_ivy/test_functional/test_nn/test_activations.py#L154)

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hi, we should test all relevant parameters for the frontend function. I've highlighted a way to do this below

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py (hunk 3 above)
Comment (PatricYan) [reply ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:45]: done, I have updated the test case

**Comment by PatricYan** [comment]
Hi, the test-docstrings and test-fronted-jax error is caused by the backend code updated, it is other test case relu errors
I will take another PR to fix the test errors of the functions

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+def test_jax_nn_softmax(
+    dtype_and_x,
+    axis,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    input_dtype, x = dtype_and_x
+    tmp = []
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:30]: You don't need to use tmp, its not necessary

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    tmp.append(x)
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:31]: don't need this

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="jax",
+        fn_tree="nn.softmax",
+        x=data,
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:43]: `x = np.asarray(x, dtype=input_dtype)`

File: ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    axis,
+    as_variable,
+    num_positional_args,
+    native_array,
+    fw,
+):
+    input_dtype, x = dtype_and_x
+    tmp = []
+    tmp.append(x)
+    data = np.asarray(tmp, dtype=input_dtype)
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_jax_nn_activations.py:32]: don't need this

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Some small changes requested. If the error is due to the helper function I will merge.

**Comment by PatricYan** [comment]
tmp is used to ensure to convert just one element to the List format. If not use it, need to add min dims 1 for the test data, or it will be failed for the dims are 0

**Review by jkeane508 — APPROVED** [review]
